ADR/0039-workspace-deprecation.md

```diff
@@ -0,0 +1,98 @@
+
+Workspace roles give a way to describe who has what rights in the workspace.
+
+Workspaces happen to be our mechanism for managing quota - a fact inherited from the [kubesaw] stack.
+
+### Some changes
+
+Some relevant changes that have occurred or are ongoing.
+
+* We want to make Konflux a successful open source projects that fits into the
+cloud native apps ecosystem. We want everything to work on vanilla Kubernetes - no OpenShift required.
+
+* We had our [first upstream Konflux release](https://github.com/konflux-ci/konflux-ci) which doesn't use [kubesaw] or relying
+on the Workspace abstraction, and is deployable on vanilla Kubernetes.
+
+* We made a decision (now long ago) to drop [KCP] from our architecture, which initially introduced the workspace concept. Both transparent multi-cluster as a feature and workspaces as a concept came from KCP.
+
+* We made the decision (at Red Hat) to stop trying to offer a managed service interface for Konflux. The user signup, waitlist, and workspace provisioning interfaces made a lot of sense there.
+
+## Decision
+
+1. We will remove the Workspace abstraction from Konflux and use standard Kubernetes namespaces
+directly for providing a place for users to run their pipelines and store their configurations
+and secrets.
+
+2. We will stop using the term Workspace, and start to use the term Namespace.
+
+3. The Konflux UI will expose a wizard for creating a new namespace. This
+wizard will be available to any user that has permissions to create namespaces.
+
+4. Konflux will provide an optional automation for creating a namespace for a user
+when he/she logins for the first time.
+
+5. Konflux won't be opinionated about the mechanism for initializing/maintaining namespaces with
+supporting resources such as ResourceQuota and LimitRange. Konflux will differ
+this responsibility to other tools which are specialized in this task such as 
+(but not limited to) [Kyverno](https://kyverno.io/policies/best-practices/add-ns-quota/add-ns-quota/) and [ArgoCD](https://github.com/konflux-ci/namespace-generator).
+
+6. Konflux will provide a thin [backend service](https://github.com/konflux-ci/workspace-manager) for listing the namespaces where the user has at least view access
+to the Konflux CRDs. This list will be used by the namespace switcher in the UI.
+This is required since the Kubernetes API doesn't let the user to list a subset
+of namespace. The user gets permissions to list all namespace or none.
+
+7. Konflux will provide ClusterRoles that will grant permissions to the Konflux
+and Tekton CRDs. Those will be (aggregated)[https://github.com/konflux-ci/konflux-ci/issues/440] to the built-in Kubernetes roles (`view`, `edit`, `admin`).
+
+8. Public viewer access will be provided by assigning the `view` role (see above) to the
```

Comment:
So I think it should be done without the abstraction, less work and maintenance.

Reply:
> So I think it should be done without the abstraction, less work and maintenance.

the amount of work for this is the same with or without the abstraction, the work is already done for KubeSaw which required considerable changes in the proxy logic, but if other implementation for single cluster support do not require proxy then this will be easier